project/CodeStone1125/README.md

```diff
@@ -0,0 +1,149 @@
+
+    The algorithm is very simple; iterate over the internal edges "diagonals" created by triangulation and remove non-essential diagonals. A non-essential diagonal is found when at either end of the diagonal the points linked would be convex. This is determined by testing the orientation of the points. Here is the pseudo code in `python`:
+
+  ```python
+  def hertel_mehlhorn(polygon):
+    convex_polygons = []  # List to store resulting convex polygons
+    
+    while len(polygon) > 3:  # Continue until only a triangle remains
+        ear_vertex = find_convex_ear_vertex(polygon)
+        if ear_vertex is None:
+            # Unable to find an ear vertex, the polygon is invalid
+            break
+        
+        # Create a new convex polygon with the ear vertex and its neighbors
+        new_convex_polygon = [ear_vertex.prev, ear_vertex, ear_vertex.next]
+        convex_polygons.append(new_convex_polygon)
+        
+        # Remove the ear vertex from the polygon
+        ear_vertex.prev.next = ear_vertex.next
+        ear_vertex.next.prev = ear_vertex.prev
+        polygon.remove(ear_vertex)
+    
+    # Add the remaining triangle to the list of convex polygons
+    convex_polygons.append(polygon)
+    
+    return convex_polygons
+
+    def find_convex_ear_vertex(polygon):
+        # Find and return an ear vertex from the polygon if one exists
+        # Implement the logic to check if a vertex is convex and has an "ear"
+        # This involves checking the orientation of the vertices.
+        # Return None if no ear vertex is found.
+
+  ```
+
+## Prospective users
+
+* Game Developer: Determine whether something in a game world is attach by a character is very common in game developing. There are even games making this as one of their main features, like [Super Mario]() and [Pac-Man]()
+
+| ![Super Mario](./pictures/Mario.png) | ![Pac-Man](./pictures/pacman.png) |
+|:-----------------------------------:|:-----------------------------------:|
+| **Super Mario** |**Pac-Man**|
+
+* Collision Prevention for Automation Equipment: Robots and automation devices used in industrial automation need to ensure that they do not collide or interfere with each other while performing tasks. 2D collision detection can monitor the positions of individual machine components to prevent unnecessary collisions, enhance production efficiency, and safeguard equipment.
+
+## System Architecture
+
+
+
+| ![](./pictures/Mario.png) |
+|:-----------------------------------:|
+| **System flow chart** |
+
+## API Description
+
+* Python API:
+  * `example(var1, var2)`: explain
+
+* C++ API:
+  * `example(var1, var2)`: explain
+
+## Engineering Infrastructure
+
+* Automatic build system: `cmake`
+* Version control: `git`
+* Testing framework: `pytest`
```

Comment:
Do you plan to wrap *all* functions to Python for testing?

Reply:
For my code, I think there would be one C++ API, and I would wrap it in Python for testing.

Reply:
It will be hard to make a fancy presentation with only a single C++ function.  It sounds a little bit oversimplification.  I suggest you to give it more thought about the numerical calculation to code.

Reply:
Sure I will modified my proposal.

Reply:
Please do not mark conversation resolved.  It makes subsequent review harder.

Reply:
Sure, I will keep it in mind.